### PR TITLE
Slash Tires

### PR DESCRIPTION
**Pull request by Rezitoo**
Added the ability for players to use Sharp Objects to Slash vehicle tires, but watch out there is a chance 911 gets a call!

Uses third eye to slash tire
Chance of missing
911 Call

### CODE REVIEW

**Comment by Rezitoo** [comment]
https://i.imgur.com/kuC6OGT.png

**Comment by Savage912-Official** [comment]
Nice!

**Comment by minipunch** [comment]
Well first attempt didn't go so well...

**Comment by minipunch** [comment]
Not sure if this is cause of your script though cause it just happened again without event trying.. all I did was open F8

**Comment by minipunch** [comment]
Don't think u have to worry about that stuff, think that is a FiveM issue

but you did forget to start `slashtires` in `server.cfg`

**Comment by minipunch** [comment]
Also like you mentioned this removes the `TAB` to put away weapon hotkey

We should indeed fix that so that still works

What was the reason for that here??

File: resources/interaction-menu/cl_thirdEye.lua [hunk 1]
```diff
@@ -1,49 +1,21 @@
...
+	["wheel_rm2"] = 47,
+	["wheel_rm3"] = 48
+}
+
 function onInteract(targetData,itemData)
     print(targetData.name,targetData.label) --> my_buffalo_target   Buffalo
     print(itemData.name,itemData.label)     --> lock_door           Lock Door
 end
 
-function onATMInteract(targetData, itemData)
```
Comment (minipunch) [inline resources/interaction-menu/cl_thirdEye.lua:6]: Looks like you merged incorrectly here

We need these functions (this is stuff added not long ago)

File: resources/interaction-menu/cl_thirdEye.lua [hunk 2]
```diff
@@ -80,6 +52,10 @@ function onVehicleOptionSelect(a, buttonInfo, hitHandle)
         end
     elseif buttonInfo.label == "Stickers" then
         TriggerEvent("rcore_stickers:open")
+    elseif buttonInfo.label == "Slash Tire" then
+        for name, index in pairs(bones) do
```
Comment (minipunch) [inline resources/interaction-menu/cl_thirdEye.lua:56]: Where is `bones` defined here?

File: resources/interaction-menu/cl_thirdEye.lua [hunk 3]
```diff
@@ -449,45 +429,8 @@ function addMechanicVehicleOptions()
     })
 end
 
-function addCivModelOptions()
```
Comment (minipunch) [inline resources/interaction-menu/cl_thirdEye.lua:452]: again lost stuff

File: resources/interaction-menu/inventory/hotkeys/cl_hotkeys.lua [hunk 4]
```diff
@@ -55,7 +55,7 @@ Citizen.CreateThread(function()
                         vehiclePlate = exports.globals:trim(vehiclePlate)
                     end
                     -- cycle through slots keys --
-                    if IsDisabledControlJustPressed(24, KEYS.TAB) then
+                    if IsDisabledControlJustPressed(24, KEYS.CTRL) then
```
Comment (minipunch) [inline resources/interaction-menu/inventory/hotkeys/cl_hotkeys.lua:58]: Why change this ?

**Review by minipunch — CHANGES_REQUESTED** [review]

**Comment by Rezitoo** [comment]
> Also like you mentioned this removes the `TAB` to put away weapon hotkey
> 
> 
> 
> We should indeed fix that so that still works
> 
> 
> 
> What was the reason for that here??

So the script requires the weapon to be in hand to actually slash the tire and with pressing tab to open the third eye to begin to slash the tire would in turn put the weapon away making slashing the tire impossible

File: resources/interaction-menu/cl_thirdEye.lua (hunk 2 above)
Comment (Rezitoo) [reply resources/interaction-menu/cl_thirdEye.lua:56]: Bones are defined at the beginning of the script https://i.imgur.com/zjr8Wlr.png

File: resources/interaction-menu/cl_thirdEye.lua (hunk 1 above)
Comment (Rezitoo) [reply resources/interaction-menu/cl_thirdEye.lua:6]: Will get it added

File: resources/interaction-menu/cl_thirdEye.lua (hunk 3 above)
Comment (Rezitoo) [reply resources/interaction-menu/cl_thirdEye.lua:452]: Will get it added

File: resources/interaction-menu/inventory/hotkeys/cl_hotkeys.lua (hunk 4 above)
Comment (Rezitoo) [reply resources/interaction-menu/inventory/hotkeys/cl_hotkeys.lua:58]: This was my attempt to change the key to put away the weapon, Is what enabled the script to work with the third eye in the first place

File: resources/interaction-menu/cl_thirdEye.lua (hunk 1 above)
Comment (DatBoiWeep) [reply resources/interaction-menu/cl_thirdEye.lua]: should be back now

File: resources/interaction-menu/cl_thirdEye.lua (hunk 3 above)
Comment (DatBoiWeep) [reply resources/interaction-menu/cl_thirdEye.lua]: should be back now

**Comment by Savage912-Official** [comment]
> > > Also like you mentioned this removes the `TAB` to put away weapon hotkey
> > > We should indeed fix that so that still works
> > > What was the reason for that here??
> > 
> > 
> > So the script requires the weapon to be in hand to actually slash the tire and with pressing tab to open the third eye to begin to slash the tire would in turn put the weapon away making slashing the tire impossible
> 
> To avoid changing the third eye hotkey we should just give the player a knife (if they have one in their inv) before we call the slash tire event and it ought to work then right

We

> > > Also like you mentioned this removes the `TAB` to put away weapon hotkey
> > > We should indeed fix that so that still works
> > > What was the reason for that here??
> > 
> > 
> > So the script requires the weapon to be in hand to actually slash the tire and with pressing tab to open the third eye to begin to slash the tire would in turn put the weapon away making slashing the tire impossible
> 
> To avoid changing the third eye hotkey we should just give the player a knife (if they have one in their inv) before we call the slash tire event and it ought to work then right

Yeah that would work

**Review by minipunch — APPROVED** [review]

**Pull request merged**